controllers/object_controls.go

```diff
@@ -1539,6 +1543,55 @@ func TransformSandboxDevicePlugin(obj *appsv1.DaemonSet, config *gpuv1.ClusterPo
 	return nil
 }
 
+// TransformDRADriverPlugin transforms nvidia-dra-driver-plugin daemonset with required config as per ClusterPolicy
+func TransformDRADriverPlugin(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n ClusterPolicyController) error {
```

Comment:
Out of scope for this PR: How much merit is there in refactoring these `Transform*` functions to strip out the common logic that is performed for all containers?

Reply:
There is definitely some merit. We currently apply some common transforms for all DaemonSets here before calling the individual `Transform*` functions: https://github.com/NVIDIA/gpu-operator/blob/58b19546c08722c5c633243c2656df3c902e06a7/controllers/object_controls.go#L718-L729

We could strip out more, but I believe the main issue is that each `Transform*` function is reading configuration from a different data type. E.g. `TransformDRADriverPlugin` reads from a struct of type `DRADriverSpec` while `TransformDriver` reads from a struct of type `DriverSpec`.